jcg_testcases/src/main/scala/JSTestExtractor.scala

```diff
@@ -0,0 +1,83 @@
+import TestCaseExtractorApp.debug
+
+import java.io.File
+import scala.io.Source
+
+object JSTestExtractor extends TestCaseExtractor {
+    val language = "js"
+
+    override def extract(inputDir: File, outputDir: File, prefixFilter: String = ""): Unit = {
+        val resources: Array[File] = getResources(new File(inputDir, language), prefixFilter)
```

Comment:
Some of these things seem common enough to extract them to a method in the parent trait. Or maybe rather have the concrete `extract` method in the parent trait that then calls a subclass method to do the actual per-file extraction?